File: app/src/main/java/wangdaye/com/geometricweather/weather/apis/OpenMeteoGeocodingApi.java
package wangdaye.com.geometricweather.weather.apis;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;
import wangdaye.com.geometricweather.weather.json.openmeteo.OpenMeteoLocationResults;

/**
 * Open-Meteo API
 */
public interface OpenMeteoGeocodingApi {

    @GET("v1/search?format=json")
    Call<OpenMeteoLocationResults> callWeatherLocation(@Query("name") String name,
                                                       @Query("count") int count,
                                                       @Query("language") String language);

    @GET("v1/search?format=json")
    Observable<OpenMeteoLocationResults> getWeatherLocation(@Query("name") String name,
                                                            @Query("count") int count,
                                                            @Query("language") String language);

}

File: app/src/main/java/org/breezyweather/weather/openmeteo/OpenMeteoGeocodingApi.kt
package org.breezyweather.weather.openmeteo

import io.reactivex.rxjava3.core.Observable
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query
import org.breezyweather.weather.openmeteo.json.OpenMeteoLocationResults

/**
 * Open-Meteo API
 */
interface OpenMeteoGeocodingApi {
    @GET("v1/search?format=json")
    fun callWeatherLocation(
        @Query("name") name: String,
        @Query("count") count: Int,
        @Query("language") language: String
    ): Call<OpenMeteoLocationResults>

    @GET("v1/search?format=json")
    fun getWeatherLocation(
        @Query("name") name: String,
        @Query("count") count: Int,
        @Query("language") language: String
    ): Observable<OpenMeteoLocationResults>
}